### ISSUE TITLE
UnexpectedAlertOpenError in IE even after handling an alert 

### ISSUE THREAD

**Issue by bharathp** [angular/protractor#3146]
Hi,

I've an alert that occurs after clicking on a link of gmail and my application continues only if I accept it.

I'm trying to accept the alert by switching into it but couldn't accept the alert. I even tried setting capability (unexpectedAlertBehaviour:'accept',).
This capability accepts my alert, but even after that I get the same error "UnexpectedAlertOpenError".

Tried in many ways by passing KEY.ENTER, ESC, etc. but unable to get focus on the alert. 
Could you please guide me how to handle such kinds of alerts using protractor.

**Comment by Harsha509** [comment]
+1

**Comment by sjelin** [comment]
Have you tried everything in https://github.com/angular/protractor/issues/308?

**Comment by bharathp** [comment]
@sjelin We've tried all the possibilities that are mentioned in #308 but unable to handle the alert.
We've performed workaround by using AutoIt to click on 'OK' button of alert through which **we are able to accept the alert now.** This was the only way we found to move forward and this worked fine for us. 
Is there any way with which we can handle this using protractor directly ?.

**Configuration:**
windows 7
protractor 2.5.1
Internet Explorer 11

**Comment by Harsha509** [comment]
Hi @sjelin,

Am facing the same issue,
We have a verification link in a mail, on clicking the @href link a new tab opens in which a modal dialog box appears on the screen and we are supposed to accept the alert to continue a registration process. We are able to accept alert using **browser.driver.switchTo().alert().accept()**  in chrome and Firefox. But where in ie, we are unable to handle it.we are having **UnexpectedAlertOpenError ** error while running tests in IE. Tried every thing and every possibility to mentioned in #308 to resolve it.

Configuration:
windows 7
protractor 2.5.1
Internet Explorer 11

Please let me know if you need more info on this.

Thanks in advance
Sri Harsha.

**Comment by inthegarage** [comment]
I am able to reproduce this bug. Possibly the best way of overcoming this is to write a script which kills the alert or the event and then execute that browser.executeScript. It does mean a little bit more test code clutter, it's the only workaround as it stands. 
Yes it will mean you don't test IE fully.

**Comment by juliemr** [comment]
@inthegarage do you have a reproducible test case? We should open up a bug with IEDriver at https://github.com/SeleniumHQ/selenium/issues.

**Comment by inthegarage** [comment]
Yes I just had an event on window.beforeonunload, that returns a string and the alert message appears.
In Chrome and FF this works perfectly (alert catch), IE it's just ignored. I'll post the code here over the weekend.

**Comment by inthegarage** [comment]
@juliemr 

So in any controller you can do the following:

```
    function checkForChanges(event, isWindowEvent) {
        // check for a null unload, when protractor tests are running.
        // the only way to get them to pass in IE.
        if (($scope.form && !$scope.form.$pristine))) {
            var txt = gettextCatalog.getString('You have unsaved changes, are you sure?');
            if (isWindowEvent) {
                return txt;
            } else {
                if (!confirm(txt)) {
                    event.preventDefault();
                }
            }
        }
    }

    $window.onbeforeunload = function (event) {
        return checkForChanges(event, true);
    };
```


In the protractor test you can do the following:

```
            browser.navigate().refresh().catch(function() {
                return browser.switchTo().alert().then(function (alert) {
                      // Is this expected?
                });
            });


The catch is never caught in IE. FF and Chrome is all OK.


```
Should this be forwarded to Selenium? cheers

**Comment by juliemr** [comment]
Sorry for the slow response - yes, this should be filed with Selenium. Thanks!
